### ISSUE TITLE
Vertex alpha not applied

### ISSUE THREAD

**Issue by artechdowon** [andyp123/blender_vertex_color_master#61]
I am currently using Blender 4.3.
There is no problem painting general vertex colors.
However, if I isolate the A channel for the alpha channel and apply changes after painting, the painted alpha value is not saved.
When I check the attribute value, I confirmed that the value changes when painting the alpha channel, but after applying, all values ​​are reset to 1.000.

When I create a new blender file and paint the alpha channel, it is saved in the attribute, but in some scenes, all alpha channel values ​​are lost.

**Comment by Andrej730** [comment]
@artechdowon attached demo appears to be broken.

Couldn't reproduce this:

> However, if I isolate the A channel for the alpha channel and apply changes after painting, the painted alpha value is not saved.
> When I check the attribute value, I confirmed that the value changes when painting the alpha channel, but after applying, all values ​​are reset to 1.000.

**Comment by artechdowon** [comment]
> @artechdowon attached demo appears to be broken.
> 
> Couldn't reproduce this:
> 
> > However, if I isolate the A channel for the alpha channel and apply changes after painting, the painted alpha value is not saved.
> > When I check the attribute value, I confirmed that the value changes when painting the alpha channel, but after applying, all values ​​are reset to 1.000.
![vertexalpha](https://github.com/user-attachments/assets/8b0c1766-82a1-490e-87cb-78e0dff54b84)

**Comment by Andrej730** [comment]
> After applying vertex alpha and saving, when I run Blender again, all the information values ​​are gone.

Alpha is not gone, it's just not very visibile - check spreadsheet for the real values in attributes.

By default vertex paint only works with RGB and to work with Alpha, VCM is starting a special mode where you paint alpha using RGB and then it's saved as just Alpha.

https://github.com/user-attachments/assets/21f44a86-a331-4298-92b8-7d9b1ad5b6a8

**Comment by artechdowon** [comment]
I tested the issues in detail.
- When Blender is running, alpha colors are displayed in the viewport when converting the A channel to isolate.
When Blender is closed and the file is opened again, the alpha color is no longer visible. (It is recorded in the spreadsheet.)

- When working on the alpha channel for other objects, the alpha is not visible. <- I wonder if it is the case when there is a certain number of vertices or more.

- When I check the spreadsheet after applying, the information is saved.

When I isolate the A channel again for modification, the alpha color is not visible. (White state)
If I apply again then all attribute A channel values ​​are overwritten with 1.000.

* In summary, the alpha channel painting value is recorded, but the alpha channel is not displayed when working for a certain period of time or when the file is opened again and isolated.
Since all values ​​are reset when reapplying, it is impossible to preserve or re-modify the existing work values.

**Comment by Andrej730** [comment]
Still don't understand what's the issue, can you show some way to reproduce it?

**Comment by Andrej730** [comment]
@artechdowon can you reproduce it consistently or it's semi-random?

I'm using Blender 4.3 and couldn't reproduce this:
https://github.com/user-attachments/assets/bcb1293f-40fb-4357-95a6-4ae3232a7b32

**Comment by artechdowon** [comment]
@Andrej730 This happens quite often. What I want is for the alpha colors I painted before to be preserved and displayed again when I isolate.

**Comment by Andrej730** [comment]
> The core of my issue is that the alpha channel that I painted before is not displayed again.
> If I keep what I did once, there is no problem. If I need to edit it again, I cannot see the previous settings, so I cannot edit it or I have to do it again.
> Can you look into this part? And thank you so much for your attention to my issue. I'm sorry if I bothered you too much.

I couldn't reproduce it (see my demo above), no idea why it happens. 
You can check system console (window - toggle system console) for errors when it happens, maybe it'll give us some clues.

**Comment by artechdowon** [comment]
Here are two demos.
A- When everything is normal, when you isolate, the values ​​you worked on before are preserved and displayed.
However, this is extremely rare.
Additional fixes are possible.
https://github.com/user-attachments/assets/e317c190-01ce-434f-bac8-92b4789144ae



B- At some point, when you isolate something you worked on again, the previous work is not displayed. (Of course, the alpha value is stored, so you can see it by connecting the alpha channel value to the color in the shader.)
This happens almost 99% of the time.
https://github.com/user-attachments/assets/3630d58a-5025-4140-957d-086076441bdc
What I want is state A.
But most of the time it ends up with the same problem as B.

**Comment by Andrej730** [comment]
I understand the issue but I've no idea what causing it since I cannot reproduce it. 
Is there a way to reproduce it consistently?

Also:
> You can check system console (window - toggle system console) for errors when it happens, maybe it'll give us some clues.

**Comment by artechdowon** [comment]
Yes, it happens pretty consistently.
It switches from A to B within a few minutes of starting Blender.
There was nothing unusual in the system console either.
After isolate, bpy.ops.vertexcolormaster.apply_isolated(discard=False)

After apply,
bpy.ops.vertexcolormaster.isolate_channel(src_channel_id='A')
That's it.

I think the color of the A channel, which should be isolated, is not displayed properly. That's the question.

**Comment by Andrej730** [comment]
> It switches from A to B within a few minutes of starting Blender.

There must me some actions during those minutes that's making it transition from A to B, it couldn't be just time Blender running.

> There was nothing unusual in the system console either.
> After isolate, bpy.ops.vertexcolormaster.apply_isolated(discard=False)

Are you sure you're checking system console and not "Info" panel? System console doesn't typically show executed operators.

**Comment by artechdowon** [comment]
Almost B state occurs immediately.
State like A is extremely difficult to see.
I was mistaken. The execution statements I uploaded were copied from Info. There was nothing strange about anything else.

**Comment by Andrej730** [comment]
> Almost B state occurs immediately.
> State like A is extremely difficult to see.

But why I cannot reproduce it then?

E.g. steps I do:
1. New blender session
2. Start vertex paint mode on the default cube.
3. Isolate Alpha.
4. Paint it.
5. Apply changes.
6. Isolate alpha again.
7. Alpha is okay, has changes from step 4.

**Comment by Andrej730** [comment]
No idea.

**Comment by artechdowon** [comment]
hey. I think I found a temporary solution.
If you select the vertices of the entire mesh before isolating the alpha channel and then isolate it, you can display the alpha colors you worked on before again.

I don't know why this happens, but I'm glad I can continue my work for now.

I think this will be a temporary solution for those who are having the same alpha channel display problem as me.
https://github.com/user-attachments/assets/ec01b969-146f-4586-8edf-7c6e314c8c23

**Comment by KingOfDranovis** [comment]
Hey, I'm having the same issue (4.3) where the alpha is inconsistent with applying, in some scenes it works fine and in others it doesn't save/apply. The solution of selecting all vertices doesn't seem to help either. I've had to fight it by editing the alpha of an object all at once, which obviously doesn't work for every object I want to work on.


> > You can check system console (window - toggle system console) for errors when it happens, maybe it'll give us some clues.

When I clicked Isolate Active Channel, I was able to get this from the system console:
 `WARN (bpy.rna): C:\Users\blender\git\blender-v430\blender.git\source\blender\python\intern\bpy_rna.cc:1373 pyrna_enum_to_py: current value '41' matches no enum in 'FloatColorAttribute', '', 'data_type'`

**Comment by KingOfDranovis** [comment]
![Image](https://github.com/user-attachments/assets/0b37f5d6-64b3-4ccf-8b18-ce01bd6b4090)

@artechdowon I think I found the issue, so check how your Color Attribute is set up.

It seems that the issue, at least for me, was because the Color Attribute was set up incorrectly. I had it as the domain Face Corner and data type Color, and the alpha wasn't applying or showing up correctly. Converting it to data type Byte Color allowed it to apply and show on subsequent channel isolations. 

When you make a Color Attribute in the Object Properties, the default settings are Vertex domain and Color data type. However, when you go into Vertex Paint _without_ a Color Attribute, the one that Blender creates automatically has Face Corner domain and Byte Color data type.
This would explain the inconsistency between scenes, if the Color Attribute was created manually it has incorrect settings for viewing the alpha value.

Provided I'm actually correct in diagnosing the issue, it should be easily reproduceable. 

Whether an edit to the code to fix this, or just a warning to the user that it's set up in a way such that the alpha wouldn't apply correctly, either should be workable.

**Comment by artechdowon** [comment]
@KingOfDranovis 
Thank you. I am aware of the issue. I will temporarily remember the storage type of the property.
